### PR TITLE
Reproducer for #8988 S1172 FP: Don't report on partial method parameters

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Reproducer for #8988

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/TestCases/MethodParameterUnused.CSharp11.cs [hunk 1]
```diff
@@ -30,3 +30,19 @@ public class MethodParameterUnused
     }
 }
 
+namespace Issue_123
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/MethodParameterUnused.CSharp11.cs:33]:
```suggestion
namespace Issue_8988
```

File: analyzers/tests/SonarAnalyzer.Test/Rules/MethodParameterUnusedTest.cs [hunk 2]
```diff
@@ -94,9 +94,33 @@ public class Sample
...
+            roslynCS.AddSnippet("""
+                using System.CodeDom.Compiler;
+
+                [GeneratedCode("TestTool", "Version")]
+                public partial class Generated
+                {
+                    private partial void M(int a, int unused);
+                }             
+                """)
+            .AddSnippet("""
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/Rules/MethodParameterUnusedTest.cs:112]: The tabbing doesn't look right

File: analyzers/tests/SonarAnalyzer.Test/TestCases/MethodParameterUnused.RoslynCfg.cs [hunk 3]
```diff
@@ -346,6 +346,19 @@ partial class MethodAssignedToActionFromPartialClass
...
+        partial void FalsePositive(string one, string two);
+    }
+
+    public partial class PartialMethod
+    {
+        partial void FalsePositive(string one, string two) // Noncompliant FP
+        {
+            Console.Write(two);
+        }
+    }
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/MethodParameterUnused.RoslynCfg.cs:360]: What about `namespace Issue_8988` ?

**Review by cristian-ambrosini-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.Test/TestCases/MethodParameterUnused.RoslynCfg.cs (hunk 3 above)
Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/MethodParameterUnused.RoslynCfg.cs:360]: I added the issue number and a link to the class name. I like to keep the test case next to the exiting partial class from above.

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**